lesson21/src/js/main.js

```diff
@@ -0,0 +1,215 @@
+    };
+
+    table.appendChild(thead).appendChild(tr);
+}
+
+const renderTableBody = usersData => {
+    const tbody = document.createElement("tbody");
+    tbody.id = "js-usersTableBody";
+
+    for(const user of usersData){
+        const fragment = document.createDocumentFragment();
+        const tr = document.createElement("tr");
+
+        Object.keys(usersTableColumn).forEach((column) => {
+            const td = createAttributedElements({
+                tag:"td",
+                valuesByAttributes:{
+                    class:"border border-gray-300 px-4 py-2"
+                },
+                str:user[column]
+            })
+            fragment.appendChild(tr).appendChild(td);
+        });
+
+        table.appendChild(tbody).appendChild(fragment);
+    }
+}
+
+const sortButtonAttributes = [
+    {dataOrder:"desc",src:"../img/asc.svg", alt:"asc-image"},
+    {dataOrder:"both",src:"../img/desc.svg",alt:"desc-image"},
+    {dataOrder:"asc",src:"../img/both.svg",alt:"both-image"}
+]
+
+const sortByUsersTableColumn = {
+    "id" : "ID"
+}
+
+const addSortButtons = (userKey,userValue,th) => {
+    for(const value of Object.values(sortByUsersTableColumn)){
+        userValue === value && th.appendChild(createSortButtons(userKey));
+    }
+}
+
+const createSortButtons = (columnKey) => {
+    const sortButtonsBox = createAttributedElements ({
+        tag:"div",
+        valuesByAttributes:{
+            id:`js-${columnKey}Buttons-Box`,
+            class: "w-10 inline-block align-middle"
+        }
+    })
+
+    const fragment = document.createDocumentFragment();
+
+    for(const button of sortButtonAttributes){
+        const sortButton = createAttributedElements({
+            tag:"button",
+            valuesByAttributes:{
+                class: `js-${columnKey}SortButton`,
+                'data-order':button.dataOrder
+            }
+        });
+
+        button.dataOrder !== "asc" && sortButton.classList.add("hidden");
+
+        const sortButtonImage = createAttributedElements({
+            tag:"img",
+            valuesByAttributes:{
+                class:"pointer-events-none",
+                src:button.src,
+                alt:button.alt
+            }
+        })
+
+        fragment.appendChild(sortButton).appendChild(sortButtonImage);
+    }
+
+    sortButtonsBox.appendChild(fragment);
+    return sortButtonsBox;
+}
+
+const toggleHiddenClassSortButton = (sortButtons,index) => {
+    sortButtons[index].classList.add("hidden");
+    (index === sortButtons.length -1) ? index = 0 : index++;
+    sortButtons[index].classList.remove("hidden");
+}
+
+const updateTableBody = usersData =>  {
+    document.getElementById("js-usersTableBody").remove();
+    renderTableBody(usersData);
+};
+
+const sortUsersData = (key, usersData, dataOrder) => {
+    const cloneUsersData = [...usersData];
+
+    const sortButtonsFunc = {
+        "both": () => {
+            updateTableBody(usersData);
+        },
```

Comment:
Why not make this function merely a content to let people sort?
 It would be better to execute `updateTableBody(usersData);` separately to separate the responsibilities.